### DISCUSSION TITLE
PERF_COUNTERS differences betwen APP_SEND_BYTES and UDP_SEND metrics

### DISCUSSION THREAD

**Discussion by akrisfx** [microsoft/msquic#4714] [Q&A] [answered]
what are the differences between this metric pairs*, under what conditions will their values differ

*pairs:
```
QUIC_PERF_COUNTER_APP_SEND_BYTES  // Total bytes sent by applications 
QUIC_PERF_COUNTER_APP_RECV_BYTES  // Total bytes received by applications 
and
QUIC_PERF_COUNTER_UDP_RECV  // Total UDP datagrams received 
QUIC_PERF_COUNTER_UDP_SEND  // Total UDP datagrams sent 
```

**Comment by nibanks** [comment, marked as answer]
Answered in https://github.com/microsoft/msquic/discussions/4713.

**Reply by akrisfx** [reply]
then what are the differences between APP_RECV_BYTES and UDP_RECV_BYTES

**Reply by nibanks** [reply]
App layer vs UDP layer.

**Reply by akrisfx** [reply]
thank you for explaining the seemingly simple things <3
